src/pages/home/report/ReportActionContextMenuItem.js

```diff
@@ -29,52 +34,100 @@ function getButtonState(isHovered = false, isPressed = false) {
 const propTypes = {
     icon: PropTypes.elementType.isRequired,
     text: PropTypes.string.isRequired,
+    successIcon: PropTypes.elementType,
+    successText: PropTypes.string,
     isMini: PropTypes.bool,
+    onPress: PropTypes.func.isRequired,
 };
 
 const defaultProps = {
     isMini: false,
+    successIcon: null,
+    successText: '',
 };
 
-const ReportActionContextMenuItem = (props) => {
-    const {getButtonStyle, getIconFillColor} = getReportActionContextMenuItemStyles(props.isMini);
-    return (
-        props.isMini
-            ? (
-                <Tooltip text={props.text}>
-                    <Pressable style={({hovered, pressed}) => getButtonStyle(getButtonState(hovered, pressed))}>
+class ReportActionContextMenuItem extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            success: false,
+        };
+        this.onPress = this.onPress.bind(this);
+    }
+
+    /**
+     * Called on button press and mark the run
+     *
+     * @memberof ReportActionContextMenuItem
+     */
+    onPress() {
+        if (this.state.success) {
+            return;
+        }
+        const pressResult = this.props.onPress();
```

Comment:
NAB, if this is returning a boolean we can just call it inside the `if ()` without creating a new variable